sql/parser/builtins.go

```diff
@@ -1145,46 +1146,63 @@ func regexpReplace(ctx EvalContext, s, pattern, to, sqlFlags string) (Datum, err
 	return DString(newString.String()), nil
 }
 
+var flagToByte = map[syntax.Flags]byte{
+	syntax.FoldCase: 'i',
+	syntax.DotNL:    's',
+}
+
+var flagToNotByte = map[syntax.Flags]byte{
+	syntax.OneLine: 'm',
+}
+
 // regexpEvalFlags evaluates the provided Postgres regexp flags in
 // accordance with their definitions provided at
 // http://www.postgresql.org/docs/9.0/static/functions-matching.html#POSIX-EMBEDDED-OPTIONS-TABLE.
 // It then returns an adjusted regexp pattern.
 func regexpEvalFlags(pattern, sqlFlags string) (string, error) {
-	goReFlags := map[rune]struct{}{'s': {}, 'm': {}}
+	flags := syntax.DotNL
 
-	for _, flag := range sqlFlags {
-		switch flag {
+	for _, sqlFlag := range sqlFlags {
+		switch sqlFlag {
 		case 'g':
-			// Ignore for now, valid flag that should be handled elsewhere.
+			// Handled in `regexpReplace`.
 		case 'i':
-			goReFlags['i'] = struct{}{}
+			flags |= syntax.FoldCase
 		case 'c':
-			delete(goReFlags, 'i')
+			flags &= ^syntax.FoldCase
 		case 's':
-			goReFlags['s'] = struct{}{}
+			flags |= syntax.DotNL
 		case 'm', 'n':
-			delete(goReFlags, 's')
-			delete(goReFlags, 'm')
+			flags &= ^syntax.DotNL
+			flags |= syntax.OneLine
 		case 'p':
-			goReFlags['s'] = struct{}{}
-			delete(goReFlags, 'm')
+			flags |= syntax.DotNL
+			flags |= syntax.OneLine
 		case 'w':
-			delete(goReFlags, 's')
-			goReFlags['m'] = struct{}{}
+			flags |= syntax.DotNL
+			flags &= ^syntax.OneLine
 		default:
-			return "", fmt.Errorf("invalid regexp flag: %q", flag)
+			return "", fmt.Errorf("invalid regexp flag: %q", sqlFlag)
 		}
 	}
 
-	if len(goReFlags) == 0 {
```

Comment:
Does it still make sense to return an unmodified pattern without the non-capturing parentheses when no flags are needed (`flags == 0`)?